mir_scenarios/mir_states/ros/src/mir_states/common/manipulation_states.py

```diff
@@ -165,6 +165,42 @@ def execute(self, userdata):
 
         return 'succeeded'
 
+
+class move_arm_and_gripper(smach.State):
+
+    def __init__(self, conf, target=None, blocking=True, tolerance=None, timeout=10.0):
+        smach.State.__init__(self,
+                             outcomes=['succeeded', 'failed'],
+                             input_keys=['move_arm_to'])
+
+        joint_names = ['arm_joint_1', 'arm_joint_2', 'arm_joint_3', 'arm_joint_4', 'arm_joint_5']
+        self.arm_moveit_client  = MoveitClient('/arm_', target, timeout, joint_names)
+        self.pub = rospy.Publisher("/gripper_controller/command", std_msgs.msg.Float64,     queue_size=1)  
+        self.conf = conf
+        #self.sub = rospy.Subsriber("/gripper_controller/state", JointState    , state_cb)                                 
+
+    def get_targets(self, group_name):
+        text = rospy.get_param('/robot_description_semantic')
+        pattern = "group_state[ \\\\\n\r\f\v\t]*name=\"" + self.conf + "\"[ \\\\\n\r\f\v\t]*group=\"arm_1_gripper\">[ \\\\\n\r\f\v\t]*<joint[ \\\\\n\r\f\v\t]*name=\"gripper_motor_left_joint\"[ \\\\\n\r\f\v\t]*value=\""
+
+        match = re.search(pattern,text)
```

Comment:
can you explain in the comments what's going on here? maybe put a snippet of what the srdf looks like